Allow ## in comments (#299)

prosoitos opened the pull request:
"##" is emacs ESS way to start a commented line

jimhester commented on R/comment_linters.R:
```diff
@@ -28,7 +28,7 @@ ops <- list(
 #' @export
 commented_code_linter <- function(source_file) {
   res <- re_matches(source_file$file_lines,
-                    rex("#", any_spaces,
```
Maybe a better solution would be `some_of("#")`, which corresponds to `#+` and would allow any number of `#`

prosoitos replied:
I thought of it, but I was worried to disrupt something else for people running lintr in RStudio or other interfaces that I don't use. So I opted for the minimal possible code change that would fix my problem. But either of these options would totally work for me :)

jimhester commented:
Looks good, thanks! Could you also add a note to NEWS.md with the change and mentioning the issue number and your GitHub username; then I can merge this.

prosoitos commented:
Will do. Thank you for all the work on lintr and for your quick help!!

prosoitos commented:
Following your advice, I replaced `one_of("#", "##")` by `some_of("#")` and I added a line to `NEWS.md`. Because my branch is now 9 commits behind, there are conflicts on that last file. Please let me know if there is anything I should be doing on my end (again, this is my first PR, so I am not used to the proper protocol).

Thank you!

jimhester commented:
Everything looks good, thanks again and congratulations on your first successful pull request!

prosoitos commented:
Thank you!
